### PR TITLE
Enable golangci-lint and fix reported errors

### PR DESCRIPTION
**Pull request by janos**
This PR fulfills go-modules PR by enabling linters that are used on master but in updated version report errors on the codebase. While go-modules PR does not make changes to the swarm application, only tooling, this one does and for easier review it is split into a separate changeset.

In go-modules PR, a deprecated gometalinter.v2 is replaced by golangci-lint as the older one does not play well with modules.

Goangci-lint configuration is added mainly for goconst linter to be more permissive on min-occurrences, as some of the changes required with default configuration are not required by my opinion. Even some strings are repeated multiple times, it does not mean that their semantics is the same justifying additional constant. This is debatable and open for discussion.

Running the complete set of goglanci-lint linters produce a very high number of errors and I am suggesting to address some of them in the future and standardize linting.

### CODE REVIEW

File: cmd/swarm-snapshot/create_test.go [hunk 1]
```diff
@@ -127,8 +127,8 @@ func TestSnapshotCreate(t *testing.T) {
 			// as strings to every node sorted services
 			sort.Strings(wantServices)
 
-			for i, n := range snap.Nodes {
-				gotServices := n.Node.Config.Services
+			for i := 0; i < len(snap.Nodes); i++ {
+				gotServices := snap.Nodes[i].Node.Config.Services
```
Comment (nolash) [inline cmd/swarm-snapshot/create_test.go:131]: Why this change?

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+linters-settings:
+  goconst:
+    # minimal length of string constant, 3 by default
+    min-len: 3
+    # minimal occurrences count to trigger, 3 by default
+    min-occurrences: 5
```
Comment (nolash) [inline .golangci.yml:6]: What exactly do these two choices mean?

Comment (janos) [reply .golangci.yml:6]: Comments are taken from the original documentation. The first one is the minimal length of the string that can be considered for a constant and the second one is the minimal number of that string occurrences in the package to be reported as constant definition required.

File: cmd/swarm-snapshot/create_test.go (hunk 1 above)
Comment (janos) [reply cmd/swarm-snapshot/create_test.go:131]: Previous iteration is coping the mutex, new one does not.

**Review by nolash — APPROVED** [review]

**Review by skylenet — APPROVED** [review]

**Comment by janos** [comment]
Thank you for reviews, I will merge this PR as soon as https://github.com/ethersphere/swarm/pull/1532 receives more approvals an is ready to be merged.

**Review by holisticode — APPROVED** [review]
